vetapp/app/controllers/visits_controller.rb

```diff
@@ -0,0 +1,53 @@
+class VisitsController < ApplicationController
+       before_action :authenticate_user!
+
+    def index
+        @animal=Animal.all
+        @visit=Visit.all
+        @user=User.all
+    end
+    
+    def show
+    end
+    
+    def new
+        @visit=Visit.new
+        @animal=Animal.all
+        @user=User.all
+    end
+    
+    def create
+        @animal=Animal.all
+        @user=User.all
+        @visit=Visit.new
+        @visit.weight=params[:visit][:weight]
+        @visit.animal_id=params[:animal_id]
+        @visit.vet=params[:visit][:vet]
+        @visit.user_id=current_user.id
+        @visit.save!
```

Comment:
If you want to assign all these values at once using `#create` (or `#update`) you can whitelist keys in the params hash -- you may have seen this pattern using a private method in the controller if you used the Rails scaffolds in the beginning and got all that boilerplate code. Something like 
```
private

def visit_params
  params.require(:visit).permit(:weight, : vet, :animal_id, :user_id)
end
```
Check out the Rails guides on strong parameters: https://guides.rubyonrails.org/action_controller_overview.html#strong-parameters